text/0028-bill-of-materials.md

```diff
@@ -0,0 +1,105 @@
+those used in the construction of that image. The availability of this data is
+a key value proposition of Cloud Native Buildpacks.
+
+As an implementation of Cloud Native Buildpacks, the Paketo project should
+support adding bill of materials metadata across the different components that
+go into an app image. This RFC serves as an intent to implement the BOM and
+defines the minimal set of key/values to be included.
+
+## Motivation
+
+A BOM can include an array of different metadata about application dependencies
+and buildpack dependencies such as SHAs, URIs, versions, license information,
+vulnerability identifiers, etc. This information is valuable for users who want
+an accurate inventory of everything in their final application image, or that
+may have been used to build it. Having access to that information makes it
+easier to audit for vulnerabilities and license information.
+
+The Paketo Java Buildpacks as well as a handful of other Paketo Buildpacks
+already make the BOM available. Another motivation for doing this work is to
+achieve a consistent user experience across the Paketo project.
+
+## Detailed Explanation
+
+The following Paketo components should have BOM metadata attached to them:
+
+### Stacks
+Stacks (such as those found in the [Paketo Stacks repository](https://github.com/paketo-buildpacks/stacks))
+should have BOM metadata that includes in-depth information on all of the OS
+level packages installed as part of the stack.
+
+The minimal set of metadata for stacks should be:
+```
+{
+  "name" : "<package name>",
+  "version" : "<package version>",
+  "arch" : "<compatible architecture>",
+  "source" : {
+    "name" : "<package source name>",
+    "version" : "<package source version>",
+    "upstreamVersion" : "<package source upstream version>"
+  },
+  "summary" : "<package summary>"
+}
+```
+This closely resembles the metadata that is already available on stacks.
+
+### Directly Installed Dependencies
+Dependencies that directly provide runtimes and/or are tools used for
+compilation should have BOM metadata surfaced about them. This includes both
+the dependencies in the final application image, as well as those used during
+the image building process. An example of this type of dependency is the
+node-engine dependency that is provided by the [Paketo Node Engine
+Buildpack](https://github.com/paketo-buildpacks/node-engine). These are the
+type dependencies that are usually listed in the
+[`buildpack.toml` file](https://github.com/paketo-buildpacks/node-engine/blob/main/buildpack.toml).
+
+The minimal set of keys to include in these type of BOM entries are:
+```
+[[bom]]
+name = "<dependency name>"
+
+[bom.metadata]
+  sha256 = "<hash of dependency artifact from uri>"
+  uri = "<uri to dependency>"
+  version = "<dependency version>"
+```
+The optional set of keys may include:
+```
+  # Optional parameters
+  cpe = "<version-specific common platfrom enumeration>"
```

Comment:
Seems like maybe we should include `purl` and `swid` as optional fields as well - https://cyclonedx.org/use-cases/#known-vulnerabilities

Reply:
@ForestEckhardt thoughts here? My thought was that between the `cpe` notation, and the source URL/SHA256, that would be plenty of information about vulnerabilities/where the package came from. 

Adding `swid`/`purl` would be quite involved to generate/get this for every dependency in the dep-server which to me might not be worth the tradeoff of the value add. It's also worth noting that tools like `syft` that generate CycloneDX can easily get this information

Reply:
Ah looking at the most recent comments on the upstream RFC, I see that CPEs aren't as encouraged as SWID/pURLs since it's going to be deprecated soon...I am rethinking my original comment now and can see that there is a value add with SWID/pURL.

Reply:
After some investigation it is unclear to me whether or not we are on the hook for generating the `SWIDs`. My current research leads me to believe that the dependency maintainer is responsible for creating a `SWID` so our ability to add this information is out of our control.

Reply:
We have decided to move forward with support package URLs following [some investigation](https://github.com/paketo-buildpacks/dep-server/issues/42#issuecomment-880961646).

Reply:
Check out https://github.com/paketo-buildpacks/rfcs/pull/57/commits/5fc173d286d60e685ed81876e1f4a1765cd39b46